modules/imgcodecs/test/test_tiff.cpp

```diff
@@ -840,6 +840,79 @@ TEST(Imgcodecs_Tiff, readWrite_predictor)
     }
 }
 
+// See https://github.com/opencv/opencv/issues/23416
+typedef std::pair<int,bool> Imgcodes_Tiff_TypeAndComp;
+typedef testing::TestWithParam< Imgcodes_Tiff_TypeAndComp > Imgcodecs_Tiff_Types;
+
+TEST_P(Imgcodecs_Tiff_Types, readWrite_alltypes)
+{
+    const int types = get<0>(GetParam());
+    const bool isCompAvailable = get<1>(GetParam());
+
+    const Mat img = cv::Mat::zeros( 120, 160, types );
+    ASSERT_FALSE(img.empty());
```

Comment:
`::zeros()` is not IO operation or subject for test. I do not see reason for the assert.

Reply:
Thank you, I'll remove it.